File: etamen-backend/app/Modules/Pharmacies/Http/Requests/CreatePharmacyOrderRequest.php
<?php

namespace App\Modules\Pharmacies\Http\Requests;

use App\Modules\Pharmacies\Domain\Enums\PharmacyDeliveryMethod;
use Illuminate\Foundation\Http\FormRequest;
use Illuminate\Validation\Rule;

class CreatePharmacyOrderRequest extends FormRequest
{
    public function authorize(): bool
    {
        return true;
    }

    public function rules(): array
    {
        return [
            'patient_user_id' => ['prohibited'],
            'payment_id' => ['prohibited'],
            'subtotal' => ['prohibited'],
            'grand_total' => ['prohibited'],
            'commission_amount' => ['prohibited'],
            'provider_net_amount' => ['prohibited'],
            'payment_status' => ['prohibited'],
            'order_status' => ['prohibited'],
            'pharmacy_provider_id' => ['required', 'integer', 'exists:providers,id'],
            'prescription_id' => ['nullable', 'integer', 'exists:pharmacy_prescriptions,id'],
            'items' => ['required', 'array', 'min:1', 'max:50'],
            'items.*.product_id' => ['required', 'integer', 'exists:pharmacy_products,id'],
            'items.*.quantity' => ['required', 'integer', 'min:1', 'max:1000'],
            'items.*.stock_quantity' => ['prohibited'],
            'items.*.unit_price' => ['prohibited'],
            'items.*.line_total' => ['prohibited'],
            'delivery_method' => ['required', Rule::in(PharmacyDeliveryMethod::values())],
            'delivery_address' => ['nullable', 'string', 'max:1000', 'required_if:delivery_method,delivery'],
            'notes' => ['nullable', 'string', 'max:1000'],
        ];
    }
}

File: etamen-flutter/lib/features/pharmacy/data/models/create_pharmacy_order_request.dart
import 'package:etamen_app/features/pharmacy/domain/entities/pharmacy_cart_item.dart';

class CreatePharmacyOrderRequest {
  const CreatePharmacyOrderRequest({
    required this.pharmacyProviderId,
    required this.items,
    this.prescriptionId,
    this.notes,
    this.deliveryAddress,
    this.deliveryMethod = 'pickup',
  });

  final int pharmacyProviderId;
  final List<PharmacyCartItem> items;
  final int? prescriptionId;
  final String? notes;
  final String? deliveryAddress;
  final String? deliveryMethod;

  Map<String, dynamic> toJson() {
    return {
      'pharmacy_provider_id': pharmacyProviderId,
      'items': items
          .map(
            (item) => {
              'product_id': item.product.id,
              'quantity': item.quantity,
            },
          )
          .toList(growable: false),
      if (prescriptionId != null) 'prescription_id': prescriptionId,
      if (notes?.trim().isNotEmpty == true) 'notes': notes!.trim(),
      if (deliveryAddress?.trim().isNotEmpty == true)
        'delivery_address': deliveryAddress!.trim(),
      if (deliveryMethod?.trim().isNotEmpty == true)
        'delivery_method': deliveryMethod!.trim(),
    };
  }
}
